toolkit/offchain/src/plutus_script.rs

```diff
@@ -18,6 +19,22 @@ impl PlutusScript {
 		Self { bytes: cbor.into(), language }
 	}
 
+	pub fn from_ogmios(ogmios_script: OgmiosScript) -> anyhow::Result<Self> {
+		if let Plutus(script) = ogmios_script {
+			let language_kind = match script.language.as_str() {
+				"plutus:v1" => LanguageKind::PlutusV1,
+				"plutus:v2" => LanguageKind::PlutusV2,
+				"plutus:v3" => LanguageKind::PlutusV3,
+				_ => {
+					return Err(anyhow!("Unsupported Plutus language version: {}", script.language))
+				},
+			};
+			return Ok(Self { bytes: script.cbor, language: language_kind });
+		} else {
+			return Err(anyhow!("Expected Plutus script, got something else."));
```

Comment:
Cosmetic: Rust is expression language, you can remove Return and semicolon if you don't make early return